src/core/symbology/qgsfillsymbollayer.cpp

```diff
@@ -3259,7 +3259,8 @@ void QgsLinePatternFillSymbolLayer::renderPolygon( const QPolygonF &points, cons
     context.setOriginalValueVariable( mDistance );
     distance = mDataDefinedProperties.valueAsDouble( QgsSymbolLayer::Property::LineDistance, context.renderContext().expressionContext(), mDistance );
   }
-  const double outputPixelDistance = context.renderContext().convertToPainterUnits( distance, mDistanceUnit, mDistanceMapUnitScale );
+  // Clip distance to a reasonable distance to avoid app freezes
+  const double outputPixelDistance = std::max( 0.025, context.renderContext().convertToPainterUnits( distance, mDistanceUnit, mDistanceMapUnitScale ) );
```

Comment:
won't this change the appearance of overlapping semi-transparent markers?

Reply:
Discussed this with @nyalldawson over chat, but for the record: 0.025 minimum still gives us 1,000 points/lines per squared painter unit, more than enough to the region to become fully opaque even with a marker with a transparency of 1/255 :)